### ISSUE TITLE
Swapping on external keyboards also affects the internal keyboards

### ISSUE THREAD

**Issue by viridityzhu** [rossmacarthur/kb-remap#4]
Hi @rossmacarthur , 
I hope you're doing well. Sorry to bother you again but I have encountered another issue and would appreciate your guidance. When using both the internal built-in keyboard and an external plugged-in keyboard, I've noticed that remapping keys on one device affects the other as well.

Here's the list of devices as shown by kb-remap --list:

```
Vendor ID  Product ID  Name
---------  ----------  ----------------------------------
0x0        0x0         Apple Internal Keyboard / Trackpad
0x0        0x0         BTM
0x0        0x0         Headset
0x4d9      0xa293      OBINS AnnePro2
0x5ac      0x0         Keyboard Backlight
```

For instance, I used kb-remap --name "OBINS AnnePro2" --swap 'option:command', and the output indicated the changes were made to the OBINS AnnePro2 keyboard:
```
Selected:
  Vendor ID: 0x4d9
  Product ID: 0xa293
  Name: OBINS AnnePro2

Applied the following modifications:
  LeftOption -> LeftCommand
  LeftCommand -> LeftOption
  RightOption -> RightCommand
  RightCommand -> RightOption
```

However, this swapping also had an unintended effect on the built-in keyboard (Apple Internal Keyboard / Trackpad). I was wondering if you could assist me in resolving this issue?

Additional information:
1. After that, I attempted to reset the built-in keyboards to their original configurations using `kb-remap --name "Apple Internal Keyboard / Trackpad" --reset`. However, this resulted in the reset of both keyboards then. 
2. Here is the output of `kb-remap --name "OBINS AnnePro2" --swap 'option:command' --dump`:
    ```
    hidutil property \
      --matching '{" "VendorID" = 0x4d9, "ProductID" = 0xa293 }' \
      --set '{"UserKeyMapping":[{"HIDKeyboardModifierMappingSrc":0x7000000e2,"HIDKeyboardModifierMappingDst":0x7000000e3},{"HIDKeyboardModifierMappingSrc":0x7000000e3,"HIDKeyboardModifierMappingDst":0x7000000e2},{"HIDKeyboardModifierMappingSrc":0x7000000e6,"HIDKeyboardModifierMappingDst":0x7000000e7},{"HIDKeyboardModifierMappingSrc":0x7000000e7,"HIDKeyboardModifierMappingDst":0x7000000e6}]}'
    ```

Thank you and best regards.

**Comment by vanderholm** [comment]
Hi there
First of all, thanks for a wonderful tool.
I have the exact same issue. Herer are the dumps of both peripherals:
```
~ % kb-remap --name "Apple Internal Keyboard / Trackpad" --swap command:option --dump

hidutil property \
  --matching '{" "VendorID" = 0x5ac, "ProductID" = 0x027e }' \
  --set '{"UserKeyMapping":[{"HIDKeyboardModifierMappingSrc":0x7000000e3,"HIDKeyboardModifierMappingDst":0x7000000e2},{"HIDKeyboardModifierMappingSrc":0x7000000e2,"HIDKeyboardModifierMappingDst":0x7000000e3},{"HIDKeyboardModifierMappingSrc":0x7000000e7,"HIDKeyboardModifierMappingDst":0x7000000e6},{"HIDKeyboardModifierMappingSrc":0x7000000e6,"HIDKeyboardModifierMappingDst":0x7000000e7}]}'

```
```
~ % kb-remap --name "PRO K/DA" --swap command:option --dump                          

hidutil property \
  --matching '{" "VendorID" = 0x46d, "ProductID" = 0xc339 }' \
  --set '{"UserKeyMapping":[{"HIDKeyboardModifierMappingSrc":0x7000000e3,"HIDKeyboardModifierMappingDst":0x7000000e2},{"HIDKeyboardModifierMappingSrc":0x7000000e2,"HIDKeyboardModifierMappingDst":0x7000000e3},{"HIDKeyboardModifierMappingSrc":0x7000000e7,"HIDKeyboardModifierMappingDst":0x7000000e6},{"HIDKeyboardModifierMappingSrc":0x7000000e6,"HIDKeyboardModifierMappingDst":0x7000000e7}]}'
```
The mappings are identical...  Is that right?

Cheers

**Comment by rossmacarthur** [comment]
Hmm looks like there is a typo in the matching thing. I haven't used this option since v0.3.0 and I think I must have broken it then. Will fix.

**Comment by rossmacarthur** [comment]
Fixed in 3cb0bea7206b1e0d803b344c429cf08588c75377, and added a test 😅 , will be released in 0.3.3 shortly

**Comment by vanderholm** [comment]
Most excellent.
I Just updated and now it is working like a charm. Many thanks
